server/pbenchinacan/deploy

```diff
@@ -1,26 +1,84 @@
+#     is printed, and the Pbench Server is deployed without a Dashboard.)
+#   - If present in the current directory, the Nginx favicon file is copied into
+#     the appropriate location in the SRV_PBENCH file system.
+#   - The ownership of the files in the SRV_PBENCH file system will are set for
+#     access inside the container.
+#   - The Pbench Server container is started and run as a detached process.
+#
+# The key inputs are the following environment variables:
+#   - PB_DEPLOY_FILES:  the location (on the host) of the configuration files
+#                       for the Pbench Server and rsyslog.
+#   - SRV_PBENCH:  the location of the file system (on the host) which will be
+#                       mounted inside the container at /srv/pbench.
+#
+# Other inputs which can be provided by environment variables include:
+#   - the name and pull policy for the Pbench Server container image
+#   - the location of the Pbench Dashboard app files and the Nginx Favicon
+#   - the location of the Pbench Git checkout (used to locate the above)
+#
+# In all cases, suitable defaults are provided for any value which is undefined.
+#
 
-# Update the favicon
-podman unshare rm /srv/pbench/public_html/favicon.ico
-podman unshare cp color-square.ico /srv/pbench/public_html/favicon.ico
+#+
+# Configuration
+#-
 
-IMAGE=images.paas.redhat.com/pbench/pbench-server:main
+# Pbench Server container image
+PB_SERVER_IMAGE=${PB_SERVER_IMAGE:-"images.paas.redhat.com/pbench/pbench-server:main"}
+PB_SERVER_IMAGE_PULL_POLICY=${PB_SERVER_IMAGE_PULL_POLICY:-"newer"}
 
-# First ensure /srv/pbench is owned by the "pbench" user INSIDE the container we run.
-podman run --rm --volume /srv/pbench:/srv/pbench:Z ${IMAGE} chown -R pbench:pbench /srv/pbench
+# Locations on the host
+PB_DASHBOARD_DIR=${PB_DASHBOARD_DIR:-${GITTOP:=$(git rev-parse --show-toplevel)}/dashboard/build}
+NGINX_FAVICON=${NGINX_FAVICON:-${GITTOP:=$(git rev-parse --show-toplevel)}/server/lib/config/color-square.ico}
+PB_DEPLOY_FILES=${PB_DEPLOY_FILES:-${HOME}/Deploy}
+SRV_PBENCH=${SRV_PBENCH:-/srv/pbench}
 
-# Deploy the actual Pbench Server instance.
+#+
+# Deployment
+#-
+
+# Update the dashboard code, removing any existing code and copying in fresh.
+# We expect that the host dashboard directory already exists and contains a
+# build of the Pbench Dashboard application; however, if it is missing, we
+# deploy the Pbench Server without a Dashboard deployment.
+podman unshare rm -r ${SRV_PBENCH}/public_html/dashboard
+if [[ -d "${PB_DASHBOARD_DIR}" ]]; then
+    podman unshare cp -r ${PB_DASHBOARD_DIR} ${SRV_PBENCH}/public_html/dashboard
+else
+    echo "Warning:  deploying the Pbench Server without a Dashboard" >&2
+    podman unshare mkdir -m 0755 -p ${SRV_PBENCH}/public_html/dashboard
+fi
+
+# Update the Nginx favicon and provide some static fallback pages
+podman unshare rm ${SRV_PBENCH}/public_html/favicon.ico
+podman unshare cp ${NGINX_FAVICON} ${SRV_PBENCH}/public_html/favicon.ico
```

Comment:
@portante, do we really need to remove the favicon file before copying in the new one?  (I grok removing the Dashboard directory tree, because we can't be sure that the new files copied in will exactly replace every old one, but here there is exactly one file which is specifically being replaced, so I don't see much gained by removing it, first.)

Reply:
@portante...ping?

Reply:
I don't think so, but I am thinking we want the favicon to be a symlink to the dashboard one.  So that this is a one-time setup.

Reply:
Using a symlink presupposes that there is a file to link to, which creates a dependency on a dashboard deployment, which creates a coupling that I'm not convinced that we want.

And, the setup isn't really "one-time", because, at least for functional test, every time is the "first time", and it seems likely that we want to adopt a similar outlook for Staging at least in part (yeah, the data might be extant, but the Server is "new" every time...right?).

Reply:
Per our discussion, we'll defer the proper sourcing of the favicon to the PR which provides the Dashboard deployment and allow that deployment to set up the Nginx favicon; in this PR, we'll just go with what I have, which is to pull it from the Git checkout.